data/src/history/manager.rs

```diff
@@ -491,6 +498,43 @@ impl Data {
 
                     true
                 }
+                Source::Internal(source::Internal::Status(status)) => {
+                    let source: source::Server;
+                    match status {
+                        source::Status::Success => {
+                            source = source::Server::new(source::server::Kind::StatusSuccess, None);
+                        }
+                        source::Status::Error => {
+                            source = source::Server::new(source::server::Kind::StatusError, None);
+                        }
+                    };
+                    if let Some(server_message) = buffer_config.server_messages.get(&source) {
+                        if !server_message.enabled {
+                            return false;
+                        }
+
+                        if let Some(seconds) = server_message.smart {
+                            let nick = match source.nick() {
+                                Some(nick) => nick.clone(),
+                                None => {
+                                    if let Some(nickname) = message.text.split(' ').nth(1) {
+                                        Nick::from(nickname)
+                                    } else {
+                                        return true;
+                                    }
+                                }
+                            };
+
+                            return !smart_filter_message(
+                                message,
+                                &seconds,
+                                most_recent_messages.get(&nick),
+                            );
+                        }
```

Comment:
Smart filtering does nothing for these messages since they're internal. No user is related to these messages.

Reply:
Maybe we need a new filter option which suppresses "duplicate" messages within some time frame?

Reply:
Hm. I was thinking of fetching the last internal message and matching against that. But maybe duplicate messages would be better. Thinking about how I'd like this smart feature to work, it would make sense to show the last errors. But then, if you see a `reconnected` but not the `disconnected` that may be odd. I was more focused on hiding all of these in the channel buffer than I was thinking how best to display these 😅 @tarkah and team, would love to hear thoughts and suggestions on this!

Reply:
Let's just do `enabled` for now then as I agree, I'm not sure another approach makes sense here. We can always add it down the road if a need arises

Reply:
I realized it didn't feel right to show nothing at all. So I added logic to the `smart` option to display if the message was received within the duration configured 🙏 Hopefully this is okay for now!